openhtf/output/web_gui/src/README.md

```diff
@@ -0,0 +1,134 @@
+# How to write a Plugin! 
+
+It is a truth universally acknowledged that an application with infinite 
+use cases must be in want of a plugin interface. Here's how to write one!
+
+## Contents
+
+A basic, bare-bones plugin package structure looks like this: 
+```
+/example_plugin
+	|
+	-- setup.py
+	|
+	-- /example_plugin
+			|
+			-- __init__.py
+			|
+			-- /plugin
+				|
+				-- main.ts
+				|
+				-- example_plugin.html
+				|
+				-- example_service.ts
+```
+
+A plugin with this structure would be able to show up in the frontend and 
+make calls to the backend Tornado server to request information.
+
+### Backend Structure
+
+Everything adding to the backend goes in the __init__.py module! It requires 
+three main parts: 
+
+#### Handler Variable
+
+This tuple is what OpenHTF will use to configure and add your tornado handler. 
+A simple example: 
+```
+handler=([r'/example/', ExampleHandler],
```

Comment:
Lists imply that it can be lengthened. For the deps portion, it makes sense since it can be any length, but for the regex/handler portion it should be a tuple. If we remove the deps like suggested, this could allow multiple handlers like so:

```py
handlers = [
  (r'/example/$', ExampleHandler),
  (r'/example/(\d+)/', IntegerHandler),
]
```